### ISSUE TITLE
Integration with nix-shell?

### ISSUE THREAD

**Issue by mjlbach** [guibou/nixGL#46]
I Just discovered nixGL when trying to figure out how to run GPU-accelerated tensorflow on non-nixos systems. I usually develop in nix-shell, and was searching for a way to append LD_LIBRARY_PATH in the shell, but didn't find anything. I came up with this shellHook.

```
shellHook = ''
export LD_LIBRARY_PATH=$(nixGLNvidia printenv LD_LIBRARY_PATH):$LD_LIBRARY_PATH
''
```

Is this the recommended solution?

**Comment by guibou** [comment]
@mjlbach that's indeed a solution. I don't know if there is a recommended solution actually. Perhaps a few nix function can be added to nixGL to ease this operation.

**Comment by apache8080** [comment]
This doesn't work if I true to use a nix pure shell. Any methods on how to allow that?

**Comment by vimpostor** [comment]
In my case exporting `$LD_LIBRARY_PATH` was not enough, my target program uses EGL and needed some of the other environment variables that `nixGL` sets.
Unfortunately there is no standalone wrapper that one could source to set those environment variables for a shellHook in a `nix-shell` (`nixGL` always uses `exec` at the end so sourcing it would `exec` in the parent context).

What worked for me in the end is this workaround:
```bash
shellHook = ''
	export $(env -i "$(which nixGL)" "$(which printenv)" | paste -sd ' ' -)
''
```

It starts `nixGL` with an empty environment, prints the environment inside, captures the stdout and reexports it outside by passing the output as separate arguments to `export`.
It works, but ultimately it would be nicer if there was a standalone sourcable script in `nixGL` itself.

Also note that the calls to `which` are necessary, `PATH` is unset due to `-i`, but shell expansion happens before that, so this is needed to find `nixGL` and `printenv` inside the `env -i`.

**Comment by considerate** [comment]
@vimpostor 

The above will overwrite your LD_LIBRARY_PATH. However, that can easily be overcome by first storing your previous LD_LIBRARY_PATH value and prefixing it onto whatever nixGL outputs, like so:
```nix
              shellHook = ''
                library_path="$LD_LIBRARY_PATH"
                export $(env -i $(which nixGL) $(which printenv))
                export LD_LIBRARY_PATH="$library_path:$LD_LIBRARY_PATH"
              '';
```

**Comment by vimpostor** [comment]
Right, but ideally there would be a separate env export script already in nixGL (maybe called `nixGLEnv`, which is called from `nixGL` and only deals with the environment variables).
That way we could avoid this workaround alltogether for `shellHook` integration. I wonder if @guibou would accept a PR for creating such a separate `nixGLEnv` script?
